feat: Implement error recovery for the parallel runner

isidentical opened the pull request:
Depends on #423

isidentical commented on src/fal/planner/schedule.py:
```diff
@@ -47,7 +47,9 @@ def create_group(
 @dataclass
 class Scheduler:
-    groups: list[TaskGroup]
+    groups: List[TaskGroup]
+    failed_groups: List[TaskGroup] = field(default_factory=list)
+    skipped_groups: List[TaskGroup] = field(default_factory=list)
```
Maybe we could add our version of `NodeStatus` and keep everything in a list. Like having an enum for `PENDING`, `SUCCEED`, `FAILED`, `SKIPPED`. Don't know though, this version seems very simple too. Any suggestions?

chamini2 replied:
> Maybe we could add our version of `NodeStatus` and keep everything in a list. Like having an enum for `PENDING`, `SUCCEED`, `FAILED`, `SKIPPED`

I would prefer this, tbh. then getting `failed_groups`/`skipped_groups` can be a function that just filters. But I am also OK with whatever you prefer.

chamini2 replied:
Are we going to change this?

chamini2 reviewed (commented):
I am kind of confused, apart from logging, what else are we achieving?

isidentical commented:
> I am kind of confused, apart from logging, what else are we achieving?

The error recovery was already partially implemented, but it was prevented from functioning since scheduler raised an error whenever something failed. That part is now removed, added logging for the failed/skipped nodes and tests.

isidentical commented on src/fal/planner/executor.py:
```diff
@@ -13,6 +13,19 @@
+    skipped_models = ", ".join(
+        model for group in skipped_groups for model in group.task.model_ids
+    )
+    logger.info(f"Failed calculating the following DBT models: {model_text}")
+    logger.info(f"Skipped calculating the following DBT models: {model_text}")
```
```suggestion
    logger.info(f"Failed calculating the following DBT models: {failed_models}")
    logger.info(f"Skipped calculating the following DBT models: {skipped_models}")
```
